tfhe/src/high_level_api/backward_compatibility/compressed_ciphertext_list.rs

```diff
@@ -6,12 +6,29 @@ use crate::{CompressedCiphertextList, Tag};
 #[derive(Version)]
 pub struct CompressedCiphertextListV0(crate::integer::ciphertext::CompressedCiphertextList);
 
-impl Upgrade<CompressedCiphertextList> for CompressedCiphertextListV0 {
+impl Upgrade<CompressedCiphertextListV1> for CompressedCiphertextListV0 {
+    type Error = Infallible;
+
+    fn upgrade(self) -> Result<CompressedCiphertextListV1, Self::Error> {
+        Ok(CompressedCiphertextListV1 {
+            inner: self.0,
+            tag: Tag::default(),
+        })
+    }
+}
+
+#[derive(Version)]
+pub struct CompressedCiphertextListV1 {
+    inner: crate::integer::ciphertext::CompressedCiphertextList,
+    tag: Tag,
+}
+
+impl Upgrade<CompressedCiphertextList> for CompressedCiphertextListV1 {
     type Error = Infallible;
 
     fn upgrade(self) -> Result<CompressedCiphertextList, Self::Error> {
         Ok(CompressedCiphertextList {
-            inner: self.0,
+            inner: crate::high_level_api::compressed_ciphertext_list::InnerCompressedCiphertextList::Cpu(self.inner),
             tag: Tag::default(),
```

Comment:
if you come from a CompressedCiphertextListV1, I think you should reuse the tag from the previous version here:
```
   tag: self.tag
```